autoscale-api/src/main/java/com/sequenceiq/periscope/api/model/DistroXAutoscaleClusterRequest.java

```diff
@@ -0,0 +1,77 @@
+package com.sequenceiq.periscope.api.model;
+
+import java.util.List;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+import com.sequenceiq.periscope.doc.ApiDescription.ClusterJsonsProperties;
+import com.sequenceiq.periscope.doc.ApiDescription.DistroXClusterJsonsProperties;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel
+public class DistroXAutoscaleClusterRequest implements Json {
+    @ApiModelProperty(ClusterJsonsProperties.ENABLE_AUTOSCALING)
+    private @NotNull Boolean enableAutoscaling;
+
+    @ApiModelProperty(DistroXClusterJsonsProperties.DISTROX_SCALING_MODE)
+    private @Valid AutoscalingModeType autoScalingMode;
+
+    @ApiModelProperty(ClusterJsonsProperties.SCALING_CONFIGURATION)
+    private @Valid ScalingConfigurationRequest scalingConfiguration;
+
+    @ApiModelProperty(ClusterJsonsProperties.TIME_ALERTS)
+    private @Valid List<TimeAlertRequest> timeAlertRequests;
+
+    @ApiModelProperty(ClusterJsonsProperties.LOAD_ALERTS)
+    private @Valid List<LoadAlertRequest> loadAlertRequests;
+
+    public DistroXAutoscaleClusterRequest() {
+    }
+
+    public DistroXAutoscaleClusterRequest(boolean enableAutoscaling) {
+        this.enableAutoscaling = enableAutoscaling;
+    }
+
+    public void setEnableAutoscaling(Boolean enableAutoscaling) {
+        this.enableAutoscaling = enableAutoscaling;
+    }
+
+    public List<TimeAlertRequest> getTimeAlertRequests() {
+        return timeAlertRequests;
+    }
+
+    public void setTimeAlertRequests(List<TimeAlertRequest> timeAlertRequests) {
```

Comment:
Would a name need to be associated with each of the cron expressions? (Can be auto-generated if required)

Reply:
name is nullable.